crates/bevy_input/src/gamepad.rs

```diff
@@ -57,3 +64,230 @@ pub enum GamepadAxisType {
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
 pub struct GamepadAxis(pub Gamepad, pub GamepadAxisType);
+
+#[derive(Default, Debug)]
+pub struct GamepadProperty {
+    pub default_button_property: ButtonProperty,
+    pub default_axis_property: AxisProperty,
+    pub default_button_axis_property: ButtonAxisProperty,
+    pub button_properties: HashMap<GamepadButton, ButtonProperty>,
+    pub axis_properties: HashMap<GamepadAxis, AxisProperty>,
+    pub button_axis_properties: HashMap<GamepadButton, ButtonAxisProperty>,
+}
+
+impl GamepadProperty {
+    pub fn get_button_properties(&self, button: GamepadButton) -> &ButtonProperty {
+        self.button_properties
+            .get(&button)
+            .unwrap_or(&self.default_button_property)
+    }
+
+    pub fn get_axis_properties(&self, axis: GamepadAxis) -> &AxisProperty {
+        self.axis_properties
+            .get(&axis)
+            .unwrap_or(&self.default_axis_property)
+    }
+
+    pub fn get_button_axis_properties(&self, button: GamepadButton) -> &ButtonAxisProperty {
+        self.button_axis_properties
+            .get(&button)
+            .unwrap_or(&self.default_button_axis_property)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct ButtonProperty {
```

Comment:
"Property" already has meaning in Bevy (ex: the `bevy_property` crate). Can we rename everything that is `XProperty` to `XSettings`?

Reply:
Renamed to `*Setting`